Quadriga/src/main/java/edu/asu/spring/quadriga/service/workbench/StatisticsSettingsService.java

```diff
@@ -0,0 +1,118 @@
+
+    @Autowired
+    private IStatisticDAO statisticDao;
+
+    @Transactional
+    public void updateRow(String projectId, String name, Boolean isChecked)
+            throws QuadrigaStorageException {
+        statisticDao.upsertStatisticsSettings(projectId, name, isChecked);
+    }
+
+    @Transactional
+    public void updateSettings(String projectId, String[] names)
+            throws QuadrigaStorageException {
+        List<String> statisticsSettingsList = getStatisticsSettingsList(projectId);
+        List<String> namesList = Arrays.asList(names);
+        if (statisticsSettingsList != null && statisticsSettingsList.size() > 0) {
+            for (String name : statisticsSettingsList) {
+                if (namesList.contains(name)) {
+                    updateRow(projectId, name, true);
+                } else {
+                    updateRow(projectId, name, false);
+                }
+            }
+        }
+    }
+
+    @Transactional
+    public List<ProjectStatisticsDTO> getStatisticsSettingsDTOList(
+            String projectId) throws QuadrigaStorageException {
+        List<ProjectStatisticsDTO> statisticsDTOList = statisticDao
+                .getStatisticsSettings(projectId);
+        return statisticsDTOList;
+    }
+
+    public List<String> getNames(List<ProjectStatisticsDTO> statisticsDTOList)
+            throws QuadrigaStorageException {
+        List<String> names = new ArrayList<String>();
+        if (statisticsDTOList != null && statisticsDTOList.size() > 0) {
+            for (ProjectStatisticsDTO psDTO : statisticsDTOList) {
+                String name = psDTO.getProjectStatisticsDTOPK().getName();
+                names.add(name);
+            }
+        }
+        return names;
+    }
+
+    @Transactional
+    public List<String> getStatisticsSettingsList(String projectId)
+            throws QuadrigaStorageException {
+        List<ProjectStatisticsDTO> statisticsDTOList = statisticDao
+                .getStatisticsSettings(projectId);
+        if (statisticsDTOList == null || statisticsDTOList.size() == 0) {
+            return new ArrayList<String>(getdefaultMapSettings().keySet());
+        }
+        List<String> names = getNames(statisticsDTOList);
+        return names;
+    }
+
+    public Map<String, Boolean> convertIntoMap(
+            List<ProjectStatisticsDTO> statisticsDTOList)
+            throws QuadrigaStorageException {
+        Map<String, Boolean> statisticsSettings = new HashMap<String, Boolean>();
+        if (statisticsDTOList != null && statisticsDTOList.size() > 0) {
+            for (ProjectStatisticsDTO psDTO : statisticsDTOList) {
+                String name = psDTO.getProjectStatisticsDTOPK().getName();
+                Boolean isChecked = psDTO.getIschecked();
+                statisticsSettings.put(name, isChecked);
+            }
+        }
+        return statisticsSettings;
+    }
+
+    public Map<String, Boolean> getdefaultMapSettings() {

```

Comment:
So, does default values should set in properties?
Should I make a new object?
or should I store in database in default table?